meorg_client/client.py

```diff
@@ -213,45 +214,67 @@ def logout(self):
             self.headers.pop("X-User-Id", None)
             self.headers.pop("X-Auth-Token", None)
 
-    def get_file_status(self, id: str) -> Union[dict, requests.Response]:
-        """Get the file status.
+    def upload_file(
+        self,
+        file_path: Union[str, list],
+        file_obj: Union[io.BufferedReader, list] = None,
+    ) -> Union[dict, requests.Response]:
```

Comment:
I think this function is larger than it needs to be. Have you considered using a list of file inputs where a file input is either a `Path` type or a `io.BufferedReader`? I.e.:
```suggestion
    def upload_file(
        self,
        file_inputs: Union[FileSpec, list[FileSpec],
    ) -> Union[dict, requests.Response]:
```
where `FileSpec = Union[str, tuple[str, io.BufferedReader]]` (`tuple` as it seems a path is still required when specifying a file object).